workflow/scripts/test_create_somalier_mqc_config.py

```
@@ -0,0 +1,239 @@
+with open(source_file, 'r') as f:
+    source_code = f.read()
+
+# Create a modified version that doesn't execute the main block
+# Stop before the "try: args = parse_args()" line
+main_try_index = source_code.find('try:\n    args = parse_args()')
+if main_try_index > 0:
+    source_code = source_code[:main_try_index]
+
+# Execute the modified source to get the functions
+exec_globals = {}
+exec(source_code, exec_globals)
```

Ruff (0.15.13)

[error] Use of `exec` detected

(S102)